File: Submissions/1189_Maximum_Number_of_Balloons.cpp
class Solution {
public:
    int maxNumberOfBalloons(string text) {
        unordered_map<char, int> frequency;
        for(int i=0; i<text.size(); ++i){
            if(text[i] != 'b' && text[i] != 'a' && text[i] != 'l' && text[i] != 'o' && text[i] != 'n') continue;
            ++frequency[text[i]];
        }
        return min({frequency['b'], frequency['a'], frequency['l']/2, frequency['o']/2, frequency['n']});
    }
};

File: Kotlin/1189_Maximum_Number_of_Balloons.kt
class Solution {
    fun maxNumberOfBalloons(text: String): Int {
        val frequency = IntArray(5)
        
        for(i in text.indices) {
            val char = text[i]
            
            when(char) {
                'b' -> ++frequency[0]
                'a' -> ++frequency[1]
                'l' -> ++frequency[2]
                'o' -> ++frequency[3]
                'n' -> ++frequency[4]
            }
        }
        
        frequency[2] /= 2
        frequency[3] /= 2
        
        return frequency.min()!!
    }
}